client-js/app/scripts/app.ts

```diff
@@ -141,13 +141,20 @@ window.addEventListener('WebComponentsReady', () => {
 
   class RegistryActivity implements Activity {
     path: Array<string>;
+    private globElem: LabradRegistry;
 
-    constructor(path: Array<string>) {
+    private api: registry.RegistryServiceJsonRpc;
+    private lifetime = new Lifetime();
+
+    constructor(api: registry.RegistryServiceJsonRpc, path: Array<string>) {
+      this.api = api;
       this.path = path;
     }
 
     start(): Promise<ActivityState> {
+      this.api.watch({path: this.path});
```

Comment:
`watch` is an api call that returns a promise. You want to make sure that the result of that promise gets dealt with, because it could be an error, and also because the activity is not fully ready to start until the call has completed. Something like:
```typescript
this.api.newItem.add(...);
return this.api.watch({path: this.path}).then(() => {
  reg.dir({path: this.path})
}).then((listing) => {
  ...
});
```